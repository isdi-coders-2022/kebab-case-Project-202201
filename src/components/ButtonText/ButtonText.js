import styled from "styled-components";
import propTypes from "prop-types";

const ButtonTxt = styled.button`
  background-color: ${(props) => props.theme.accent};
  border-color: ${(props) => props.theme.accentAlt2};
  color: white;
  border-radius: 4px;
  width: 65px;
  height: 35px;
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  font-size: 14px;
  padding: 4px 8px 4px 6px;
`;

const ButtonText = ({ text, actionOnClick, title }) => {
  return (
    <ButtonTxt type="button" onClick={actionOnClick} title={title}>
      {text}
    </ButtonTxt>
  );
};

ButtonText.propTypes = {
  text: propTypes.string,
  onclick: propTypes.func,
};
export default ButtonText;
