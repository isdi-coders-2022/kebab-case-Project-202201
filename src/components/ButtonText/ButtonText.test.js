import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import TestRenderer from "react-test-renderer";
import ButtonText from "./ButtonText";

describe("Given a ButtonText Component", () => {
  describe("When it is clicked", () => {
    test("It should do an action", () => {
      const action = jest.fn();

      render(<ButtonText actionOnClick={action} />);

      const button = screen.queryByRole("button");
      userEvent.click(button);

      expect(action).toHaveBeenCalled();
    });
  });
  describe("When is rendered", () => {
    test("It should render a button", () => {
      render(<ButtonText />);

      const button = screen.getByRole("button");

      expect(button).toBeInTheDocument();
    });

    test("It should render a a given text", () => {
      const text = "Add";

      render(<ButtonText text={text} />);

      const button = screen.getByRole("button");

      expect(button).toHaveTextContent("Add");
    });
  });

  describe("When it get a const button that creates a ButtonText", () => {
    test("then it should create and compare a snapshot with ButtonText component", () => {
      const button = TestRenderer.create(<ButtonText />);
      expect(button.toJSON()).toMatchSnapshot();
    });
  });
});
