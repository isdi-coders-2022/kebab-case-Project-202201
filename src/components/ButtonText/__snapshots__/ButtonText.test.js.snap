// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a ButtonText Component When it get a const button that creates a ButtonText then it should create and compare a snapshot with ButtonText component 1`] = `
<button
  className="sc-bdvvtL jbnYpL"
  type="button"
/>
`;
